odoo/addons/base/tests/test_ir_default.py

```diff
@@ -106,3 +106,45 @@ def test_removal(self):
         # delete the record, and check the presence of the default value
         title.unlink()
         self.assertEqual(IrDefault.get_model_defaults('res.partner'), {})
+
+    def test_multi_company_defaults(self):
+        """Check defaults in multi-company environment."""
+        company_a = self.env["res.company"].create({"name": "C_A"})
+        company_b = self.env["res.company"].create({"name": "C_B"})
+        company_a_b = (company_a + company_b)
+        company_b_a = (company_b + company_a)
+        multi_company_user = self.env['res.users'].create({
+            'name': 'u2', 'login': 'u2',
+            'company_id': company_a.id,
+            'company_ids': company_a_b.ids,
+        })
+        IrDefault = self.env["ir.default"].with_user(multi_company_user)
+        IrDefault.with_context(allowed_company_ids=company_a.ids).set(
+            'res.partner', 'ref', 'CADefault', user_id=True, company_id=True)
+        IrDefault.with_context(allowed_company_ids=company_b.ids).set(
+            'res.partner', 'ref', 'CBDefault', user_id=True, company_id=True)
+        self.assertEqual(
+            IrDefault.get_model_defaults('res.partner')['ref'],
+            'CADefault',
+            "Can't retrieve the created default value for user.",
+        )
+        self.assertEqual(
+            IrDefault.with_context(allowed_company_ids=company_a.ids).get_model_defaults('res.partner')['ref'],
+            'CADefault',
+            "Can't retrieve the created default value for user.",
```

Comment:
We could just remove the message from all the asserEqual, it's not really needed ^^.